addons/stock/models/stock_inventory.py

```diff
@@ -313,8 +314,11 @@ def _get_inventory_lines_values(self):
                 'location_id': location_id,
                 'package_id': package_id
             }
-            line_values['product_uom_id'] = self.env['product.product'].browse(product_id).uom_id.id
+            product_ids.add(product_id)
             vals.append(line_values)
+        product_id_to_product = dict(zip(product_ids, self.env['product.product'].browse(product_ids)))
```

Comment:
I did not realize it in the 14.0 PR but are we sure the set order is preserved through the `zip()` ?

Reply:
Yeah you're right, I missed that. I replaced the set by an OrderedSet to ensure the order is preserved.

For the PR that got merged into 14.0 it should be fine as I'm zipping a recordset and a list
https://github.com/odoo/odoo/blob/26bdebf6fd8b77013ea449a02184e8a843ffe9e9/addons/stock/models/stock_inventory.py#L455-L456

But I also made that mistake in the 13.0 PR odoo/odoo#80106. Thankfully it has not been merged yet.